lib/src/html/html_generator_instance.dart

```diff
@@ -21,52 +21,52 @@ import 'templates.dart';
 
 class HtmlGeneratorInstance implements HtmlOptions {
   final HtmlGeneratorOptions _options;
-
-  String get url => _options.url;
   final Templates _templates;
-  final Package package;
-  final Directory out;
-  final List<ModelElement> documentedElements = <ModelElement>[];
+  final Package _package;
+  final String _outputDirectoryPath;
+  final List<ModelElement> _documentedElements = <ModelElement>[];
   final StreamController<File> _onFileCreated;
+
   @override
   String get relCanonicalPrefix => _options.relCanonicalPrefix;
+
   @override
   String get toolVersion => _options.toolVersion;
-  String get faviconPath => _options.faviconPath;
-  bool get useCategories => _options.useCategories;
-  bool get prettyIndexJson => _options.prettyIndexJson;
+
+  String get _faviconPath => _options.faviconPath;
+  bool get _useCategories => _options.useCategories;
 
   // Protect against bugs in canonicalization by tracking what files we
   // write.
   final Set<String> writtenFiles = new Set<String>();
 
-  HtmlGeneratorInstance(this._options, this._templates, this.package, this.out,
-      this._onFileCreated);
+  HtmlGeneratorInstance(this._options, this._templates, this._package,
+      this._outputDirectoryPath, this._onFileCreated);
 
   Future generate() async {
-    if (!out.existsSync()) out.createSync();
```

Comment:
This directory is already created in entry point

Reply:
optional: remove it from the entry point instead?  Might save work for other callers in the future.

Reply:
Maybe higher up in the call hierarchy. I like that this file doesn't have to worry about `Directory` instances.